worker/worker.go

```diff
@@ -158,11 +158,14 @@ func (w *Worker) Transmit(addr string, id string, responseTo string, metadata ma
 	return
 }
 
-// EmitEvent emits a WorkerEvent, worker message id, and an optional message.
+// EmitEvent emits a WorkerEvent, worker message id, and key-value pairs of optional data.
 func (w *Worker) EmitEvent(event ipc.WorkerEventName, messageID string, message string) error {
```

Comment:
This method signature might as well change to accept a `data map[string]string` parameter instead of `message string`.